test/ao/sparsity/test_data_scheduler.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Owner(s): ["module: unknown"]
+
+import logging
+from torch.ao.sparsity import BaseDataScheduler, DataNormSparsifier
+
+from torch.testing._internal.common_utils import TestCase
+from torch import nn
+import torch
+
+logging.basicConfig(format='%(asctime)s - %(name)s - %(levelname)s - %(message)s', level=logging.INFO)
+
+
+class ImplementedDataScheduler(BaseDataScheduler):
+    def __init__(self, sparsifier, sparsifier_hyperparam, last_epoch=-1, verbose=False):
+        super().__init__(sparsifier, sparsifier_hyperparam, last_epoch, verbose)
+
+    def get_schedule_param(self):
+        pass
+
+
+class TestBaseDataScheduler(TestCase):
+    def _get_data(self):
+        tensor1, param1, emb1 = torch.randn(5, 5), nn.Parameter(torch.randn(10, 10)), nn.Embedding(50, 5)
+        data_list = [
+            ('tensor1', tensor1), ('param1', param1), ('emb1', emb1)
+        ]
+        defaults = {
+            'sparsity_level': 0.7,
+            'sparse_block_shape': (1, 4),
+            'zeros_per_block': 2
+        }
+        data_with_config = [
+            {
+                'name': 'tensor2', 'data': torch.randn(4, 4),
+                'config': {'sparsity_level': 0.3}
+            }
+        ]
+        return data_list, defaults, data_with_config
+
+    def _get_sparsifier(self):
+        data_list, defaults, data_with_config = self._get_data()
+        sparsifier = DataNormSparsifier(data_list, **defaults)
+        for data_config_dict in data_with_config:
+            name, data, config = data_config_dict['name'], data_config_dict['data'], data_config_dict['config']
+            sparsifier.add_data(name=name, data=data, **config)
+        return sparsifier
+
+    def _get_scheduler(self, sparsifier):
+        schedule_param = self._get_scheduler_param()
```

Comment:
nit: it's not obvious that `_get_scheduler` depends on `_get_scheduler_param`, can we make scheduler param as an argument to this function to make it more explicit?

Reply:
makes sense - made changes